metadata/src/main/java/org/apache/kafka/migration/MigrationRecoveryState.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.migration;
+
+import java.util.Objects;
+
+/**
+ * Persistent state needed to recover an ongoing migration. This data is stored in ZooKeeper under the "/migration"
+ * ZNode and is recovered by the active KRaft controller following an election. The absence of this data in ZK indicates
+ * that no migration has been started.
+ */
+public class MigrationRecoveryState {
```

Comment:
Does it make sense to have a `ZkControllerState` object contained in this object?

seems like the fields here are a superset of the fields in `ZkControllerState`